components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/nats/CoreListener.java

```
@@ -0,0 +1,270 @@
+        try {
+            KeyManagerFactory keyManagerFactory = null;
+            if (StringUtils.isNotEmpty(tlsConnection.getKeyStoreLocation())) {
+                KeyStore keyStore = loadKeyStore(tlsConnection.getKeyStoreType(), tlsConnection.getKeyStoreLocation(),
+                        tlsConnection.getTrustStorePassword());
+                keyManagerFactory = KeyManagerFactory.getInstance(
+                        tlsConnection.getKeyManagerAlgorithm().equals("") ?
+                                NatsConstants.DEFAULT_TLS_ALGORITHM :
+                                tlsConnection.getKeyManagerAlgorithm());
+                keyManagerFactory.init(keyStore, tlsConnection.getKeyStorePassword().toCharArray());
+            }
+
+            KeyStore trustStore = loadKeyStore(tlsConnection.getTrustStoreType(), tlsConnection.getTrustStoreLocation(),
+                    tlsConnection.getTrustStorePassword());
+            TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(
+                    tlsConnection.getTrustManagerAlgorithm().equals("") ?
+                            NatsConstants.DEFAULT_TLS_ALGORITHM :
+                            tlsConnection.getTrustManagerAlgorithm());
+            trustManagerFactory.init(trustStore);
+
+            SSLContext sslContext = SSLContext.getInstance(tlsConnection.getProtocol().equals("") ?
+                    Options.DEFAULT_SSL_PROTOCOL :
+                    tlsConnection.getProtocol());
+            sslContext.init(keyManagerFactory == null ? null : keyManagerFactory.getKeyManagers(), trustManagerFactory.getTrustManagers(),
+                    new SecureRandom());
+            return sslContext;
+        } catch (KeyStoreException | IOException | CertificateException | NoSuchAlgorithmException | UnrecoverableKeyException | KeyManagementException e) {
+            log.error("Invalid TLS parameters. Establishing connection without TLS if possible.", e);
```

Shall we throw the exceptions to the above method?